src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,122 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useMemo, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
+
+enum SortBy {
+  name = 'name',
+  sex = 'sex',
+  born = 'born',
+  died = 'died',
+}
 
 export const PeoplePage = () => {
+  const [persons, setPersons] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sortBy = searchParams.get('sort');
+  const sortOrder = searchParams.get('order');
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(response => setPersons([...response]))
+      .catch(() => setError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    let visiblePeople = [...persons];
+
+    if (sex === 'm') {
+      visiblePeople = persons.filter(person => person.sex === 'm');
+    }
+
+    if (sex === 'f') {
+      visiblePeople = persons.filter(person => person.sex === 'f');
+    }
```

Comment:
```suggestion
    if (sex) {
      visiblePeople = persons.filter(person => person.sex === sex);
    }
```
looks like we can simplify it